src/icalendar/tools.py

```diff
@@ -1,29 +1,60 @@
+"""Utility functions for icalendar."""
+
 from __future__ import annotations
+
 import random
 from datetime import date, datetime, tzinfo
 from string import ascii_letters, digits
+from warnings import warn
 
 from icalendar.parser_tools import to_unicode
 
+from .error import WillBeRemovedInVersion7
+
 
 class UIDGenerator:
-    """If you are too lazy to create real uid's."""
+    """Use this only if you're too lazy to create real UUIDs.
+    
+    .. deprecated:: 6.2.1
+
+        Use the Python standard library's :func:`uuid.uuid4` instead.
+    """
 
     chars = list(ascii_letters + digits)
 
     @staticmethod
-    def rnd_string(length=16):
-        """Generates a string with random characters of length."""
+    def rnd_string(length=16) -> str:
+        """Generates a string with random characters of length.
+
+         .. deprecated:: 6.2.1
+
+             Use the Python standard library's :func:`uuid.uuid4` instead.
+        """
+        warn(
+            "Use https://docs.python.org/3/library/uuid.html#uuid.uuid4 instead.",
+            WillBeRemovedInVersion7,
+            stacklevel=1
+        )
         return "".join([random.choice(UIDGenerator.chars) for _ in range(length)])
 
     @staticmethod
     def uid(host_name="example.com", unique=""):
-        """Generates a unique id consisting of:
-            datetime-uniquevalue@host.
-        Like:
+        """Generates a unique ID consisting of ``datetime-uniquevalue@host``.
+
+        For example:
+            
             20050105T225746Z-HKtJMqUgdO0jDUwm@example.com
+        
+        .. deprecated:: 6.2.1
+
+            Use the Python standard library's :func:`uuid.uuid5` instead.
```

Comment:
Why do we use uuid5 here and below instead of uuid4, as we did in the earlier docstrings?

Reply:
uuid5 is closer to the parameters. It gets influenced by the domain, too.

Reply:
Should we change uuid4 to uuid5? Just want to make sure that the difference is deliberate and not a typo.

Reply:
This is not a version thing of what is better/later: uuid4 is random, uuid5 uses certain information to generate the random value with a hash so you can see if they are from the same 'source'/domain.

Reply:
I understand the differences of the two functions and other uuid functions as well. My question is whether the difference here was a deliberate choice or a typo. It sounds like the former.